packages/amplify-category-api/src/index.ts

```diff
@@ -235,6 +238,11 @@ export const executeAmplifyHeadlessCommand = async (context: $TSContext, headles
       await getCfnApiArtifactHandler(context).createArtifacts(await validateAddApiRequest(headlessPayload));
       break;
     case 'update':
+      const resourceName = await getAppSyncApiResourceName(context);
+      if (!(await checkAppsyncApiResourceMigration(context, resourceName, true))) {
+        printer.error('Update operations only work on migrated projects. Run "amplify update api" and opt for migration.');
+        exitOnNextTick(0);
+      }
       await getCfnApiArtifactHandler(context).updateArtifacts(await validateUpdateApiRequest(headlessPayload));
```

Comment:
Does validateUpdateApiRequest return a bool ?

Reply:
it return a payload , other wise will throw error , if the payload doesn't matches schema